certbot-nginx/certbot_nginx/parser.py

```diff
@@ -506,8 +520,30 @@ def _add_directives(block, directives, replace):
     """
     for directive in directives:
         _add_directive(block, directive, replace)
+    if block and '\n' not in block[-1]:  # could be "   \n  " or ["\n"] !
+        block.append(nginxparser.UnspacedList('\n'))
+
+
+REPEATABLE_DIRECTIVES = set(['server_name', 'listen', 'include'])
+COMMENT = ' managed by Certbot'
+COMMENT_BLOCK = [' ', '#', COMMENT]
+
+
+def _comment_directive(block, location):
+    """Add a comment to the end of the line at location."""
+    next_entry = block[location + 1] if location + 1 < len(block) else None
+    if isinstance(next_entry, list) and next_entry:
+        if COMMENT in next_entry[-1]:
```

Comment:
Probably for maximum defensiveness we might also test that `next_entry` is longer than that and that `next_entry[-2] == "#"` or something?